mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala

```diff
@@ -31,7 +31,8 @@ import org.apache.spark.sql.DataFrame
  * @param predictionAndObservations an RDD of (prediction, observation) pairs.
  */
 @Experimental
-class RegressionMetrics(predictionAndObservations: RDD[(Double, Double)]) extends Logging {
+class RegressionMetrics(predictionAndObservations: RDD[(Double, Double)])
+  extends Logging with Serializable {
```

Comment:
Does this have to be Serializable?

If so, I'm not quite sure how extending Serializable interacts with the lazy val.  It would be useful to know when that lazy val gets instantiated (hopefully just once when the object is serialized).